trunk/wc_yotpo.php

```diff
@@ -103,30 +104,80 @@ function wc_yotpo_uninstall() {
 		delete_option('yotpo_settings');	
 	}	
 }
-function wc_yotpo_show_widget() {		 
+// REVIEWS WIDGET
+function wc_yotpo_show_reviews_widget() {		 
 	global $product;
-	if($product->get_reviews_allowed() == true) {	
-		$product_data = wc_yotpo_get_product_data($product);
-		$yotpo_div = "<div class='yotpo yotpo-main-widget'
-	   				data-product-id='".$product_data['id']."'
-	   				data-name='".$product_data['title']."' 
-	   				data-url='".$product_data['url']."' 
-	   				data-image-url='".$product_data['image-url']."' 
-	  				data-description='".$product_data['description']."' 
-	  				data-lang='".$product_data['lang']."'
-                    data-price='".$product->get_price()."'
-                    data-currency='".get_woocommerce_currency()."'></div>";
-		echo $yotpo_div;
+	if($product->get_reviews_allowed() == true) {
+		echo generate_reviews_widget_code();
+	}						
+}
+function generate_reviews_widget_code() {
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$reviews_widget_id = $yotpo_settings['reviews_widget_id'];
+	if (!$use_v3_widgets) {
+		return generate_v2_reviews_widget_code();
+	}
+	return $reviews_widget_id ? generate_v3_reviews_widget_code($reviews_widget_id) : '';
+}
+function generate_v2_reviews_widget_code() {
+	global $product;
+	$product_data = wc_yotpo_get_product_data($product);
+	return "<div class='yotpo yotpo-main-widget'
+		data-product-id='".$product_data['id']."'
+		data-name='".$product_data['title']."' 
+		data-url='".$product_data['url']."' 
+		data-image-url='".$product_data['image-url']."' 
+		data-description='".$product_data['description']."' 
+		data-lang='".$product_data['lang']."'
+		data-price='".$product->get_price()."'
+		data-currency='".get_woocommerce_currency()."'
+	></div>";
+}
+function generate_v3_reviews_widget_code($reviews_widget_id) {
+	global $product;
+	$product_data = wc_yotpo_get_product_data($product);
+	return "<div class='yotpo-widget-instance'
+		data-yotpo-instance-id='".$reviews_widget_id."'
+		data-yotpo-product-id='".$product_data['id']."'
+		data-yotpo-name='".$product_data['title']."'
+		data-yotpo-url='".$product_data['url']."'
+		data-yotpo-image-url='".$product_data['image-url']."'
+		data-yotpo-price='".$product->get_price()."'
+		data-yotpo-currency='".get_woocommerce_currency()."'
+		data-yotpo-description='".$product_data['description']."'
+	></div>";
+}
+// Q&A WIDGET
+function wc_yotpo_show_qna_widget() {		 
+	global $product;
+	$yotpo_settings = get_option('yotpo_settings',wc_yotpo_get_degault_settings());
+	$use_v3_widgets = $yotpo_settings['use_v3_widgets'];
+	$qna_widget_id = $yotpo_settings['qna_widget_id'];
```

Comment:
Getting the widget ID should be part of the `generate_v3_qna_widget_code` method.